### DISCUSSION TITLE
Graf-visningen i Wikidata, förklaringar eller dokumentation?

### DISCUSSION THREAD

**Discussion by gjonor** [salgo60/NOSAD-POC-Wikidata#3] [Wikidata]
Jag har undrat ibland vad de streckade linjerna och lila bubblorna innebär när man klickar på ett objekt i graf-visningen för frågeresultat i Wikidata. Finns det någon dokumentation för funktionerna i graf-visningen för Wikidata?

![image](https://user-images.githubusercontent.com/41752788/192161393-06cea172-a8d5-41a1-9095-c4dfa98ca039.png)

**Comment by salgo60** [comment]
Lite här [att läsa](https://en.wikibooks.org/wiki/SPARQL/Views#Graph) du kan ange själv vilka färger dom skall ha.... Jag autogenererar ofta dom graferna med en [feature EasyQuery länk](https://github.com/salgo60/SkansenWikidata/issues/9) se [video](https://www.youtube.com/watch?v=cY1C2VqIOpg)

Bra fråga jag skickade den [vidare](https://www.wikidata.org/wiki/Wikidata:Project_chat#Graph_documentation_-_defaultView:Graph) till dom smarta gubbarna
**UPDATE**: en av de smartare gubbarna [User:Tagishsimon](https://wikidata.wikiscan.org/user/Tagishsimon) pekade på samma [dokumentation](https://www.wikidata.org/wiki/Wikidata:SPARQL_query_service/Wikidata_Query_Help/Result_Views#Graph) som jag gjorde ovan

Tror man kan utläsa färg från [SPARQL koden](https://query.wikidata.org/#%23defaultView%3AGraph%0ASELECT%20%3Fnode%20%3FnodeLabel%20%3FnodeImage%20%3FchildNode%20%3FchildNodeLabel%20%3FchildNodeImage%20%3Frgb%20WHERE%20%7B%0A%20%20%7B%0A%20%20%20%20BIND%28wd%3AQ2013%20AS%20%3Fnode%29%0A%20%20%20%20%3Fnode%20%3Fp%20%3Fi.%0A%20%20%20%20OPTIONAL%20%7B%20%3Fnode%20wdt%3AP18%20%3FnodeImage.%20%7D%0A%20%20%20%20%3FchildNode%20%3Fx%20%3Fp.%0A%20%20%20%20%3FchildNode%20rdf%3Atype%20wikibase%3AProperty.%0A%20%20%20%20FILTER%28STRSTARTS%28STR%28%3Fi%29%2C%20%22http%3A%2F%2Fwww.wikidata.org%2Fentity%2FQ%22%29%29%0A%20%20%20%20FILTER%28STRSTARTS%28STR%28%3FchildNode%29%2C%20%22http%3A%2F%2Fwww.wikidata.org%2Fentity%2FP%22%29%29%0A%20%20%7D%0A%20%20UNION%0A%20%20%7B%0A%20%20%20%20BIND%28%22EFFBD8%22%20AS%20%3Frgb%29%0A%20%20%20%20wd%3AQ2013%20%3Fp%20%3FchildNode.%0A%20%20%20%20OPTIONAL%20%7B%20%3FchildNode%20wdt%3AP18%20%3FchildNodeImage.%20%7D%0A%20%20%20%20%3Fnode%20%3Fx%20%3Fp.%0A%20%20%20%20%3Fnode%20rdf%3Atype%20wikibase%3AProperty.%0A%20%20%20%20FILTER%28STRSTARTS%28STR%28%3FchildNode%29%2C%20%22http%3A%2F%2Fwww.wikidata.org%2Fentity%2FQ%22%29%29%0A%20%20%7D%0A%20%20OPTIONAL%20%7B%0A%20%20%20%20%3Fnode%20wdt%3AP18%20%3FnodeImage.%0A%20%20%20%20%3FchildNode%20wdt%3AP18%20%3FchildNodeImage.%0A%20%20%7D%0A%20%20SERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22sv%22.%20%7D%0A%7D)

<img width="1026" alt="image" src="https://user-images.githubusercontent.com/14206509/192164730-dfb99d5c-6da2-4b86-8ca1-0fa82ea8e159.png">

**Reply by gjonor** [reply]
Ok, tyvärr inte begripligt för mig i nuläget vad det där betyder, men tack ändå.

**Reply by salgo60** [reply]
@gjonor  Hittar du exempel på dashed lines så skrik till

Färgen på boxen sätter man med koden dvs. det hag gjorde nedan där jag tar partiets färg (inser fel i logiken för folk som blir vilde som ofta saknar färg eller för folk som hoppar mellan partierna)

**Reply by gjonor** [reply]
Streckade linjer och lila färg kanske visar att det är ytterligare egenskaper/kopplingar för det markerade objektet som inte härrör från fråge-resultatet.
Det borde ju vara inbyggt i graf-visningen på något sätt.

**Reply by gjonor** [reply]
Det kanske var en trivial fråga för den som är van användare, jag insåg det nu när jag tittade på ditt exempel med svenska statsministrar och regenter, att man förstås vill visa i grafen primärt de fält man valt ut i frågan.

Jag gissar att källkoden för vyerna ligger här.
https://github.com/wikimedia/wikidata-query-gui/tree/master/wikibase/queryService/ui/resultBrowser

**Reply by salgo60** [reply]
Frågan ovan med regenten och kungarna är inte triviala om du frågar mig jag bara knyckte engelska exemplet.... Han som svara på Tagishsimon är extremt duktig SPARQL finns en "SPARQL fråge jour" där han hjälpte mig på [Wikidata:Request_a_query](https://www.wikidata.org/wiki/Wikidata:Request_a_query) med en [query optimering](https://www.wikidata.org/wiki/Wikidata:Request_a_query#Outdoor_gyms_in_Berlin_(Q64)) jag aldrig [skulle ha listat ut](https://w.wiki/5kff)... Andra bra ställen att lära sig mera SPARQL
* Lucas Werkmeister körde förut en [hel del Live SPARQL](https://www.youtube.com/watch?v=EA9sLTEdcSo) och även han är i sin egen liga

**Reply by gjonor** [reply]
Jag anade att det kunde bli ett syftningsfel i det jag skrev, vi pratar nog om varandra lite. Jag menade att min frågeställning om symboler i graf-visningen kanske var trivial för en van Wikidata/SPARQL-användare, jag hade inte insett att symbolerna indikerar huruvida elementen ingår i SPARQL-resultatet eller är en produkt av interaktiv navigering i grafen.

**Comment by salgo60** [comment]
@gjonor här en fråga https://w.wiki/5k7o där jag söker ut Riksdagsmotioner som har ett ämne som har med spårtrafik [wd:Q350783](https://www.wikidata.org/wiki/Q350783?uselang=sv) att göra och vem/vilka som skrev den....

där visas gubbarna och gummorna i sin partifärg 
<img width="1026" alt="image" src="https://user-images.githubusercontent.com/14206509/192165496-4375c62f-f85e-42f8-b783-736f1eb5c4fa.png">

* [video om detta](https://youtu.be/jbd9YGNt1ek)

## SPARQL
Hämtar partifärgen - denna kod fungerar nog bra om politikerna inte byter parti eller blir vildar ;-) 
* man bör nog även kolla när en motion skrevs och kolla vilket parti en person tillhörde då...

<img width="751" alt="image" src="https://user-images.githubusercontent.com/14206509/192211990-c281893e-c40e-4a62-a274-0f81153d3f3a.png">

**Reply by salgo60** [reply]
Mer text / exempel om motioner [hittar du här](https://github.com/salgo60/NOSAD-POC-Wikidata/blob/main/Kopplingssprint%202022.md)

**Reply by salgo60** [reply]
Coolt graph exempel jag hittade för Brittiska kungahuset se [tweet](https://twitter.com/piecesofuk/status/1574706315980685312?s=20&t=Kgx_5mne-Pea0AtU5ZeMig) som jag översatt till [Sverige SPARQL query](https://query.wikidata.org/embed.html#%23defaultView%3AGraph%0A%23title%3A%20Swedish%20Monarchs%20vs.%20Prime%20Ministers%0ASELECT%20DISTINCT%20%3Fpm%20%3FpmLabel%20%3FpmImage%20%3Fmonarch%20%3FmonarchLabel%20%3FmonarchImage%20WITH%20{%0A%20%20%0A%23%20not%20sure%20if%20there%20is%20a%20more%20straightforward%20way%20to%20access%20British%20monarchs%20(this%20query%20runs%20second)%0ASELECT%20DISTINCT%20%3Fmonarch%20%3Fmonarch_start%20%3Fmonarch_end%20WHERE%20{%0A%20%20INCLUDE%20%25start_Sw%20.%0A%20%20VALUES%20%24MONARCH%20%20{%20wd%3AQ1268572%20%20%20}%0A%20%20%3Fmonarch%20wdt%3AP31%20wd%3AQ5%20.%0A%20%20%3Fmonarch%20p%3AP39%20%3Fph%20.%0A%20%20%3Fph%20ps%3AP39%20%24MONARCH%20.%0A%20%20%3Fph%20pq%3AP580%20%3Fmonarch_start%20.%0A%20%20OPTIONAL%20{%20%3Fph%20pq%3AP582%20%3Fmonarch_end%20}%0A%20%20FILTER%20(%3Fmonarch_start%20%3E%3D%20%3Fstart_Sw%20||%20%3Fmonarch_end%20%3E%3D%20%3Fstart_Sw)%0A%20%20BIND%20(%20NOW()%20AS%20%3Ftoday%20)%20%20%20%0A%20%20BIND(COALESCE(%3Fmonarch_end%2C%20%3Ftoday)%20as%20%3Fmonarch_end)%20%23%20handle%20missing%20end%20dates%2C%20hopefully%20it's%20just%20the%20current%20monarch%0A}%20}%20AS%20%25monarchs%0A%0A%23%20Find%20the%20start%20date%20of%20the%20first%20Prime%20Minister%20so%20we%20can%20filter%20monarchs%20that%20reigned%20after%20this%20point%20(this%20query%20runs%20first)%20%0AWITH%20{%20SELECT%20%3Fstart_Sw%20WHERE%20{%0A%20%20%3Fpm%20p%3AP39%20%3Fpmph_stmt%20.%0A%20%20%3Fpmph_stmt%20ps%3AP39%20wd%3AQ687075%20.%0A%20%20%3Fpmph_stmt%20pq%3AP580%20%3Fstart_Sw%20.%0A}%20%0AORDER%20BY%20%3Fstart_Sw%0ALIMIT%201%20}%20AS%20%25start_Sw%0A%0A%23%20this%20query%20runs%20last%20and%20uses%20the%20select%20in%20line%202%3B%20it%20produces%20the%20visualisation%20specified%20in%20line%201%0AWHERE%20{%0A%20%20INCLUDE%20%25monarchs%0A%20%20%3Fpm%20wdt%3AP31%20wd%3AQ5%20.%20%0A%20%20%3Fpm%20p%3AP39%20%3Fnode%20.%0A%20%20%3Fnode%20ps%3AP39%20wd%3AQ687075%20.%0A%20%20OPTIONAL%20{%20%3Fnode%20pq%3AP580%20%3Fpm_start%20}%0A%20%20OPTIONAL%20{%20%3Fnode%20pq%3AP582%20%3Fpm_end%20}%0A%20%20BIND%20(%20NOW()%20AS%20%3Ftoday%20)%0A%20%20BIND(COALESCE(%3Fpm_end%2C%20%3Ftoday)%20as%20%3Fpm_end)%20%23%20handle%20missing%20end%20dates%2C%20hopefully%20it's%20just%20the%20current%20president%0A%20%20FILTER(%20(%3Fmonarch_start%20%3E%3D%20%3Fpm_start%20%26%26%20%3Fmonarch_start%20%3C%3D%20%3Fpm_end)%20||%20(%3Fmonarch_end%20%3E%3D%20%3Fpm_start%20%26%26%20%3Fmonarch_end%20%3C%3D%20%3Fpm_end%20)%20||%0A%20%20%20%20%20%20%20%20%20%20(%3Fpm_start%20%3E%3D%20%3Fmonarch_start%20%26%26%20%3Fpm_start%20%3C%3D%20%3Fmonarch_end)%20||%20(%3Fpm_end%20%3E%3D%20%3Fmonarch_start%20%26%26%20%3Fpm_end%20%3C%3D%20%3Fmonarch_end%20))%0A%23%20get%20images%2C%20use%20a%20default%20image%20if%20none%20found%0A%20%20OPTIONAL%20{%20%3Fmonarch%20wdt%3AP18%20%3FmonarchImage%20}%0A%20%20BIND(COALESCE(%3FmonarchImage%2C%20%3Chttp%3A%2F%2Fcommons.wikimedia.org%2Fwiki%2FSpecial%3AFilePath%2FImage_needed%2C_please_upload_to_commons.wikimedia.org.jpg%3E)%20as%20%3FmonarchImage)%0A%20%20OPTIONAL%20{%20%3Fpm%20wdt%3AP18%20%3FpmImage%20}%0A%20%20BIND(COALESCE(%3FpmImage%2C%20%3Chttp%3A%2F%2Fcommons.wikimedia.org%2Fwiki%2FSpecial%3AFilePath%2FImage_needed%2C_please_upload_to_commons.wikimedia.org.jpg%3E)%20as%20%3FpmImaage)%0A%20%20SERVICE%20wikibase%3Alabel%20{%20bd%3AserviceParam%20wikibase%3Alanguage%20%22sv%2Cen%22.%20}%0A})

<img width="807" alt="image" src="https://user-images.githubusercontent.com/14206509/192553808-207dff50-7c49-4160-a1e2-062ab6d75980.png">

**Reply by gjonor** [reply]
Intressanta exempel.
